### DISCUSSION TITLE
javascript modules

### DISCUSSION THREAD

**Discussion by magray** [scottoffen/grapevine#139] [Q&A]
I'm trying to serve up a html site using javascript modules, but it won't run the javascript. How do a serve them with a  
 `Content-Type` header `text/javascript` ??
Thanks

**Comment by scottoffen** [comment]
If I understand correctly, you are asking how to serve static files using a specific content-type value, correct?

**Reply by magray** [reply]
Hello,

Yes, I am serving up a HTML file with a script tag eg. <script src="./myscript.js" type="module"></script>

but an error in the web console reads
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream". Strict MIME type checking is enforced for module scripts per HTML spec.

If I remove the type="module" from the script tag, the site runs. But I am using javascript files with exported functions. 

I can't see how grapevine can serve up this type of site.

Thanks

**Reply by scottoffen** [reply]
Thanks for the clarification. Let me see what I can do about that.

**Reply by martinwinch** [reply]
I have the same issue. I have upgraded our application from Grapevine 3.x to 5.0. Previously .js files were served up with Content-Type "application/javascript", but now we see "application/octet-stream". I think this can be fixed by adding the following to ContentType:

public static ContentType Js = new ContentType("application/javascript", false, "UTF-8");
